fuel/datasets/cifar10.py

```diff
@@ -33,23 +34,12 @@ class CIFAR10(H5PYDataset):
     filename = 'cifar10.hdf5'
 
     def __init__(self, which_set, **kwargs):
-        if which_set not in ('train', 'test'):
-            raise ValueError("CIFAR10 only has a train and test set")
-
         kwargs.setdefault('load_in_memory', True)
-        self.which_set = which_set
-
-        super(CIFAR10, self).__init__(self.data_path, which_set,
-                                      **kwargs)
+        super(CIFAR10, self).__init__(self.data_path, which_set, **kwargs)
+        self.default_transformers += (
```

Comment:
It might be hard to do consistently, since datasets sometimes rely on constructor arguments to define the default transformers (e.g. `H5PYDataset` sets `Flatten` as the default transformer if `flatten is True`, which is only known when it is instantiated).